xonsh/completer.py

```diff
@@ -163,448 +36,15 @@ def complete(self, prefix, line, begidx, endidx, ctx=None):
-        else:
-            csc = builtins.__xonsh_env__.get('CASE_SENSITIVE_COMPLETIONS')
-            startswither = startswithnorm if csc else startswithlow
-            attrlow = attr.lower()
-            opts = [o for o in dir(val) if startswither(o, attr, attrlow)]
-        prelen = len(prefix)
-        for opt in opts:
-            # check whether these options actually work (e.g., disallow 7.imag)
-            try:
-                builtins.__xonsh_execer__.eval('{0}.{1}'.format(expr, opt), _ctx)
-            except:  # pylint:disable=bare-except
-                continue
-            a = getattr(val, opt)
-            rpl = opt + '(' if callable(a) else opt
-            # note that prefix[:prelen-len(attr)] != prefix[:-len(attr)]
-            # when len(attr) == 0.
-            comp = prefix[:prelen - len(attr)] + rpl
-            attrs.add(comp)
-        return attrs
-
-
-OPTIONS_PATH = os.path.expanduser('~') + "/.xonsh_man_completions"
-SCRAPE_RE = re.compile(r'^(?:\s*(?:-\w|--[a-z0-9-]+)[\s,])+', re.M)
-INNER_OPTIONS_RE = re.compile(r'-\w|--[a-z0-9-]+')
-
-
-class ManCompleter(object):
-    """Helper class that loads completions derived from man pages."""
-
-    def __init__(self):
-        self._load_cached_options()
-
-    def __del__(self):
-        try:
-            self._save_cached_options()
-        except Exception:
-            pass
-
-    def option_complete(self, prefix, cmd):
-        """Completes an option name, basing on content of man page."""
-        csc = builtins.__xonsh_env__.get('CASE_SENSITIVE_COMPLETIONS')
-        startswither = startswithnorm if csc else startswithlow
-        if cmd not in self._options.keys():
-            try:
-                manpage = subprocess.Popen(
-                    ["man", cmd], stdout=subprocess.PIPE,
-                    stderr=subprocess.DEVNULL)
-                # This is a trick to get rid of reverse line feeds
-                text = subprocess.check_output(
-                    ["col", "-b"], stdin=manpage.stdout)
-                text = text.decode('utf-8')
-                scraped_text = ' '.join(SCRAPE_RE.findall(text))
-                matches = INNER_OPTIONS_RE.findall(scraped_text)
-                self._options[cmd] = matches
-            except:
-                return set()
-        prefixlow = prefix.lower()
-        return {s for s in self._options[cmd]
-                if startswither(s, prefix, prefixlow)}
-
-    def _load_cached_options(self):
-        """Load options from file at startup."""
-        try:
-            with open(OPTIONS_PATH, 'rb') as f:
-                self._options = pickle.load(f)
-        except:
-            self._options = {}
-
-    def _save_cached_options(self):
-        """Save completions to file."""
-        with open(OPTIONS_PATH, 'wb') as f:
-            pickle.dump(self._options, f)
+                res = out
+                lprefix = len(prefix)
+            if res is not None and res != empty_set:
```

Comment:
Isn't it better to test if `len(res) > 0`, rather than creating an extra set instance?